cmd/copy.go

```diff
@@ -1166,6 +1373,12 @@ func init() {
 	cpCmd.PersistentFlags().BoolVar(&raw.withSnapshots, "with-snapshots", false, "include the snapshots. Only valid when the source is blobs.")
 	cpCmd.PersistentFlags().StringVar(&raw.legacyInclude, "include", "", "only include these files when copying. "+
 		"Support use of *. Files should be separated with ';'.")
+	cpCmd.PersistentFlags().StringVar(&raw.includePath, "include-path", "", "only include these paths when copying. "+
+		"Supports use of * considering the relative path of items (from the root of the search.) ex. myFolder/*.txt\n;*/subDirName/*.pdf"+
```

Comment:
Ah I see that you are also supporting patterns in the path flags, I thought we wouldn't, but it actually looks good, since we enable more specific targeting.

Reply:
Oh! I must have misunderstood then. Glad it was a happy accident.

Reply:
I just realized there might be an issue with including sub-directories:

ex: source is /usr/root, and the user wants to include a specific set of sub-directories:
1. dir1 (has files directly under it)
2. dir2 (has more sub-dirs under it)

for dir1, the user can do --include-path="dir1/*", and it will match to every file under it. However, for dir2, how does the customer specify their intention? "dir2/*" only matches to the files directly under it right? Missing out the other files like "dir2/subdir/bla.txt".

Reply:
To work around this issue, and make the flag simpler, perhaps we can disallow \*s in the path flags, and let users specify relative paths to files and directories. In our implementation, we use a listTraverser to go through each one.

Reply:
Dropped wildcard implementation, correcting help text